### ISSUE TITLE
Functions created by macros in `nalgebra` such as `::zeros()` will not autocomplete or syntax highlight

### ISSUE THREAD

**Issue by CleanCut** [intellij-rust/intellij-rust#5007]
## Environment

* **IntelliJ Rust plugin version:** 0.2.117.2862-193-nightly
* **Rust toolchain version:** rustc 1.41.0 (5e1a79984 2020-01-27)
* **IDE name and version:** IntelliJ IDEA 2019.3.2 Ultimate Edition
* **Operating system:** macOS Mojave 10.14.6

## Problem description

Functions created by macros in `nalgebra` such as `::zeros()` will not autocomplete or syntax highlight

## Steps to reproduce

In `Cargo.toml`:

```toml
[dependencies]
nalgebra = "0.19.0"
```

`src/main.rs`

```rust
use nalgebra::{base::storage::Owned, Matrix, MatrixMN, Vector2, U1, U2};

fn main() {
    // No auto-completion/syntax highlighting for `zeros()` through a chain of type aliases
    let _v = Vector2::<f32>::zeros();
    // ditto with only one level of type alias
    let _v2 = MatrixMN::<f32, U2, U1>::zeros();
    // ditto without the type alias at all
    let _v3 = Matrix::<f32, U2, U1, Owned<f32, U2, U1>>::zeros();
}
```

`::zeros()` and other functions created in the same macro will not autocomplete or syntax-highlight:

<img width="900" alt="image" src="https://user-images.githubusercontent.com/5838512/75097542-2346f000-5569-11ea-9f0a-0c1304df0186.png">

# Notes:

Here is the chain of type aliases:
```Rust
pub struct Matrix<N: Scalar, R: Dim, C: Dim, S> { ... }
pub type MatrixMN<N, R, C> = Matrix<N, R, C, Owned<N, R, C>>;
pub type VectorN<N, D> = MatrixMN<N, D, U1>;
pub type Vector2<N> = VectorN<N, U2>;
```

- The struct is in [matrix.rs](https://github.com/rustsim/nalgebra/blob/dev/src/base/matrix.rs#L76)
- The type aliases are in [alias.rs](https://github.com/rustsim/nalgebra/blob/dev/src/base/alias.rs)
- The macro for `zeros()` is in [construction.rs](https://github.com/rustsim/nalgebra/blob/dev/src/base/construction.rs#L415)

**Comment by Undin** [comment]
There are two issue here:
* `impl` block is generated by macro call. The plugin supports such `impl` blocks only with new experimental macro engine. See #3628 for more details
* `impl` block is created for type alias. It's known issue #2778 and we even have PR that fixes it #4936 when there are only few type aliases for type. Otherwise (when we try to process all type aliases for given type) it leads to significant performance degradation. At this moment we don't know how to avoid it.
Unfortunately, `nalgebra` provides a lot of type aliases for `Matrix` struct. So #4936 won't fix described issue

**Comment by CleanCut** [comment]
@Undin Thanks!  Sounds like this is known, being worked on, and difficult to solve.  Thank you (all) for your hard work on this plugin. It makes my life using & teaching Rust much easier. ❤️
